### ISSUE TITLE
Header language

### ISSUE THREAD

**Issue by spaceotter** [Sarcasm/irony-mode#283]
I have some header files with the .h extension that are C++. I'm getting errors from flycheck that could be explained if irony were loading .h files as C. Is this the case and if so how can I change the behavior?

**Comment by Sarcasm** [comment]
did you configure emacs to consider `.h` as `C++`?

    (add-to-list 'auto-mode-alist '("\\.h\\'" . c++-mode))

**Comment by spaceotter** [comment]
I did try that, but it doesn't seem to affect the syntax checker. Note the code compiles fine and the compile commands file is set up. Here are some of the spurious errors:
```
   45   6 error           too few template arguments for class template 'queue' (irony)
   46   6 error           too few template arguments for class template 'queue' (irony)
   47   6 error           too few template arguments for class template 'queue' (irony)
   52  20 error           variable has incomplete type 'std::string' (aka 'basic_string<char>') (irony)
   53  26 error           variable has incomplete type 'const std::string' (aka 'const basic_string<char>') (irony)
  105  29 error           invalid operands to binary expression ('std::stringstream' (aka 'int') and 'const char *') (irony)
  108  26 error           invalid operands to binary expression ('std::stringstream' (aka 'int') and 'const char *') (irony)
  111  55 error           member reference base type 'std::stringstream' (aka 'int') is not a structure or union (irony)
  112  52 error           member reference base type 'std::stringstream' (aka 'int') is not a structure or union (irony)
  120  11 error           incomplete definition of type 'std::shared_ptr<Foo>' (irony)
  124  43 error           member reference base type 'std::stringstream' (aka 'int') is not a structure or union (irony)
  125  40 error           member reference base type 'std::stringstream' (aka 'int') is not a structure or union (irony)
  141  37 error           'std::chrono::milliseconds' (aka 'duration<int64_t, milli>') is an incomplete type (irony)
```
queue, string, iostream, chrono, and memory are explicitly included now. I had thought since they were indirectly included through a .h file it was interpreting it as C but I'm not sure now.

**Comment by Sarcasm** [comment]
I'm not sure to get it. Aren't you just missing some headers then?

**Comment by spaceotter** [comment]
Sorry, I didn't mention that the errors stayed there after the headers were added.

**Comment by Sarcasm** [comment]
Hum, well I don't know then.
You should try to compile with `clang++ -fsyntax-only` on the command line to see if there is some errors.

**Comment by spaceotter** [comment]
I ran `clang++ -fsyntax-only <copy all the stuff from irony cdb> main.cpp` and there is one (not spurious) warning, in another file. I fixed that file and it produces no output now. I had to modify the code so the error(s) are different now. In spite of clang not producing any errors, flycheck is still claiming `no matching constructor for initialization of 'Foo' (irony)`. Foo takes a std::string but is called with a literal string. It is declared in `Foo.h`.
